common/source/docs/common-buttons.rst

```diff
@@ -4,23 +4,51 @@
 Buttons
 =======
 
-Up to four button inputs to the autopilot may be defined that will execute an :ref:`Auxiliary Function <common-auxiliary-functions>` as if an RC channel switch had been used to execute it. The button input can be either a switch to ground, or a PWM input signal.
+Up to four external buttons or switches connected to the autopilot can be configured to trigger :ref:`Auxiliary Functions <common-auxiliary-functions>` similar to how RC channel switches can trigger them. These button inputs can be either analog voltage inputs (pull-to-ground or pull-to-high) or PWM inputs.
 
-:ref:`BTN_ENABLE<BTN_ENABLE>` must be set to 1 to enable this functionality and see the ``BTN`` parameters.
+.. note::
 
-Setup
-=====
+    Some :ref:`auxiliary function <common-auxiliary-functions>` have a middle position (e.g "Parachute 3 Pos Switch") but the middle position cannot be triggered from a button
 
-A GPIO pin (see :ref:`common-gpios`) can be designated as a button input by setting one of the four ``BTN_PINx`` parameters to that GPIO pins' number, eg. :ref:`BTN_PIN1<BTN_PIN1>` = 51. Up to four pins can be assigned to buttons.
+Basic Setup
+-----------
 
-These pins will have an internal pullup assigned to them automatically, requiring only a switch to ground to change state from high to low. Open or high on the pin will correspond to the ``RCx_OPTION`` function it controls as being HIGH. Ground or logic low on the pin corresponds to LOW. There is no middle option.
+For autopilots with an IOMCU (e.g. Pixhawk, CubeOrange) the external button or switch should be connected to the autopilot's AUX outputs (not the MAIN outputs).  To ease configuration and avoid timer conflicts it may be best to use AUX outputs at the high end (e.g. AUX5 or AUX6) far from outputs used for motors and servos.  For autopilots without an IOMCU (e.g. pixracer, etc) the main outputs may be used.
 
-Alternatively, a PWM signal can be used on the pin for switching, just like the PWM value for an RC channel option switch, by setting bit 0 of the button's ``BTNx_OPTIONSx<BTN_OPTIONSx>`` parameter. PWM values >1800us represent HIGH, and <1200us represent LOW.
+- Set :ref:`BTN_ENABLE<BTN_ENABLE>` = 1 (Enable) and refresh parameters
+- If using 4.2 (or later) set :ref:`SERVOx_FUNCTION<SERVO14_FUNCTION>` = -1 (GPIO) where "x" is the servo output channel connected to the button (e.g. SERVO14_FUNCTION is for AUX output 6)
+- If using 4.1 (or earlier) on an autopilot with an IOMCU (e.g. Pixhawk, CubeOrange) set BRD_PWM_COUNT low enough so that the servo output may be used for reading an analog voltage (aka GPIO) or PWM input.  E.g. BRD_PWM_COUNT = 4 means only AUX outputs 5 and 6 can be used, BRD_PWM_COUNT = 0 means AUX outputs 1 to 6 can be used.
```

Comment:
```suggestion
- In order to use an normal SERVOx output as a button input for firmware version 4.1 (or earlier) on an autopilot with an IOMCU (e.g. Pixhawk, CubeOrange), set BRD_PWM_COUNT low enough so that the servo output may be used for reading an analog voltage (aka GPIO) or PWM input.  E.g. BRD_PWM_COUNT = 4 means only AUX outputs 5 and 6 can be used, BRD_PWM_COUNT = 0 means AUX outputs 1 to 6 can be used.
```

Reply:
Same here, let's keep it simple and just focus on the aux outs.  Hopefully at some point someone will come through and re-test with other random GPIO pins and update the param descriptions and this wiki page at the same time (maybe I will..).